### PR TITLE
Turn translation validation off on timeouts.

### PR DESCRIPTION
**Pull request by ysiraichi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104464
* #104467

Follow-up to PR: #97964

After the introduction of translation validation, (TV) a few TIMM and TorchBench benchmarks
started failing due to TIMEOUT. This PR turns TV off for them.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by ysiraichi** [comment]
The remaining error seems to be coming from the base commit: `9f7ad25c987b8b69ff3a0c2aabd55a7c032de7d5`.

**Comment by ysiraichi** [comment]
Forcing merge, since [as mentioned before](https://github.com/pytorch/pytorch/pull/104464#issuecomment-1620013404) this failure is not due to this commit.

**Comment by eellison** [comment]
This leads to a 50s regression on TIMM inductor_dynamic.

**Comment by eellison** [comment]
Can we revert ?

**Comment by ysiraichi** [comment]
#104654 (currently merging) replaces the changes of this PR by something more robust. So, I would say we could just wait for that.

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -3197,16 +3232,28 @@ def detect_and_mark_batch(t):
...
-                explain=args.explain,
-                tag=args.tag,
+            # Set translation validation on by default on CI accuracy runs.
+            ci = CI(
+                args.backend,
+                training=args.training,
+                dynamic=args.dynamic_shapes,
+                device=device,
             )
+            translation_validation = args.only not in CI_TV_OFF[ci]
```
Comment (ysiraichi) [inline benchmarks/dynamo/common.py:3242]: This condition is actually incorrect. I should also take into account `args.accuracy` and `args.ci`.
